component/src/main/java/io/siddhi/extension/io/file/FileHandler.java

```diff
@@ -0,0 +1,456 @@
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.siddhi.extension.io.file;
+
+import io.siddhi.annotation.Example;
+import io.siddhi.annotation.Extension;
+import io.siddhi.annotation.Parameter;
+import io.siddhi.annotation.util.DataType;
+import io.siddhi.core.config.SiddhiAppContext;
+import io.siddhi.core.exception.SiddhiAppCreationException;
+import io.siddhi.core.exception.SiddhiAppRuntimeException;
+import io.siddhi.core.stream.ServiceDeploymentInfo;
+import io.siddhi.core.stream.input.source.Source;
+import io.siddhi.core.stream.input.source.SourceEventListener;
+import io.siddhi.core.util.config.ConfigReader;
+import io.siddhi.core.util.snapshot.state.State;
+import io.siddhi.core.util.snapshot.state.StateFactory;
+import io.siddhi.core.util.transport.OptionHolder;
+import io.siddhi.extension.io.file.util.Constants;
+import io.siddhi.extension.util.Utils;
+import io.siddhi.query.api.exception.SiddhiAppValidationException;
+import org.apache.commons.io.monitor.FileAlterationListener;
+import org.apache.commons.io.monitor.FileAlterationMonitor;
+import org.apache.commons.io.monitor.FileAlterationObserver;
+import org.apache.commons.vfs2.FileObject;
+import org.apache.commons.vfs2.FileSystemException;
+import org.apache.log4j.Logger;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.ReentrantLock;
+
+import static io.siddhi.extension.io.file.util.Util.getFileHandlerEvent;
+import static io.siddhi.extension.util.Constant.STATUS_DONE;
+import static io.siddhi.extension.util.Constant.STATUS_NEW;
+import static io.siddhi.extension.util.Constant.STATUS_PROCESS;
+import static io.siddhi.extension.util.Constant.STATUS_REMOVE;
+
+
+/**
+ * Implementation of siddhi-io-file source.
+ */
+@Extension(
+        name = "fileeventlistener",
+        namespace = "source",
+        description = "" +
+                "Fileeventlistener provides the functionality for user to get the details of files " +
+                "which have been created or modified or deleted in the execution time" ,
+
+        parameters = {
+                @Parameter(
+                        name = "uri",
+                        description =
+                                "Used to specify a file to be processed. \nThe 'uri'  should be provided.\n" +
```

Comment:
Add "this parameter is" to the description.

Reply:
fixed in fixed in https://github.com/siddhi-io/siddhi-io-file/pull/82/commits/c0a6a18523e13ed38407608e9900ada8de2d106f